cpg_workflows/targets.py

```diff
@@ -121,14 +141,30 @@ def target_id(self) -> str:
         """Unique target ID"""
         return self.name
 
+    def create_dataset(self, name: str) -> 'Dataset':
+        """
+        Create a dataset and add it to the cohort.
+        """
+        if name in self._datasets_by_name:
+            logging.debug(f'Dataset {name} already exists in the MultiCohort')
+            return self._datasets_by_name[name]
+
+        if name == self.analysis_dataset.name:
+            ds = self.analysis_dataset
+        else:
+            ds = Dataset(name=name)
+
+        self._datasets_by_name[ds.name] = ds
+        return ds
+
     def get_cohorts(self, only_active: bool = True) -> list['Cohort']:
         """
         Gets list of all cohorts.
         Include only "active" cohorts (unless only_active is False)
         """
         cohorts = list(self._cohorts_by_name.values())
         if only_active:
-            cohorts = [c for c in cohorts if c.active and c.get_datasets()]
+            cohorts = [c for c in cohorts if c.active]
```

Comment:
What is the difference between an `active` and `inactive` cohort? Related, should we even allow work on `inactive` cohorts?

Reply:
So... no. The Target ABC itself includes an Active attribute - a SG can be active or inactive, but it doesn't make sense for a Dataset, and it's not ever changed from the default (True) in Cohorts. There is some talk of deprecating Cohorts, and where/if a pipeline should fail if you query for inactive cohorts

Reply:
Just wondering then if the check should be moved into the gql, instead of getting everything and then filtering here. Fetch and process less data that way.

Reply:
Yup! I think that's where we'll end up going, but Cohort's don't currently have an 'active' attribute we can use in queries. I'm tempted to remove this `only_active` parameter in this method completely as it's meaningless, but if Cohorts do start having an active/inactive status, it probably shouldn't be up to `cpg-flow` to unilaterally decide that inactive cohorts should crash the pipeline